scripts/ansible/roles/linux/intel/tasks/redhat/sgx-packages-install.yml

```diff
@@ -44,10 +44,38 @@
   register: install_sgx
   until: install_sgx is success
 
-- name: Install the Intel DCAP packages
+- name: Download Intel SGX DCAP Packages
+  get_url:
+    url: "{{ intel_dcap_packages_url }}"
+    dest: "/tmp/sgx_linux_dcap_packages.tgz"
```

Comment:
There's a trick that can be done to make get_url and subsequent actions idempotent and more reliable. Some advantages include:
* avoid renaming the package and losing the versioning information.
* skip tasks that don't need to be reran (for idempotency)

For the get_url module you can register the return value to be used. 
To get the value of wherever the file was downloaded, you can access the return value's `dest` key. For example if you registered the return value like below, then you can refer to it in as get_dcap_package.dest
```get_url:
    url: "{{ intel_dcap_packages_url }}"
    dest: "/tmp/"
    mode: 0755
    timeout: 120
  retries: 3
  when: flc_enabled|bool
  register: get_dcap_package
```

This is also useful when there is a subsequent step that uses the file, such as the unarchive step below. You can see if the download occurred, and if so, then run the unarchive step using get_dcap_package.dest as the src.
```
- name: Unarchive Intel DCAP local rpm repo tgz
  unarchive:
    src: "{{ get_dcap_package.dest }}"
    dest: "{{ intel_dcap_packages_directory }}"
    remote_src: yes
  when: 
    - get_dcap_package.changed
    - get_dcap_package.dest
```

So if the download didn't occur because the file existed, then it will skip the unarchiving step.

There is a caveat to this, that if the system is somehow in the state where the file is downloaded, but not extracted, then the unarchive step will be skipped. You can get around this by adding a playbook variable to force it. For example:
```
- name: Unarchive Intel DCAP local rpm repo tgz
  unarchive:
    src: "{{ get_dcap_package.dest }}"
    dest: "{{ intel_dcap_packages_directory }}"
    remote_src: yes
  when: get_dcap_package.changed and get_dcap_package.dest or
             force_dcap_package_unarchive | bool
```

Reply:
Wow, very useful information, thanks! Each instance of the run should be on a clean system/the previous run should clean up, per the last step, so I think the 

```
  when: 
    - get_dcap_package.changed
    - get_dcap_package.dest
```

check should suffice.

Reply:
Make sure you access the key `dest` instead, so it's `path: {{ get_dcap_package.dest }}`
But yes the variable can be used by other tasks, so long as the tasks are after the registering of `get_dcap_package`

Reply:
Yup, learnt this the hard way 😄